x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/utils/indices.ts

```diff
@@ -28,13 +29,22 @@ import {
 export function isConnectorIndex(
   index: ElasticsearchIndexWithIngestion | undefined
 ): index is ConnectorIndex {
-  return !!(index as ConnectorIndex)?.connector;
+  const connectorIndex = index as ConnectorIndex;
+  return (
+    !!connectorIndex?.connector &&
+    connectorIndex.connector.service_type !== ENTERPRISE_SEARCH_CONNECTOR_CRAWLER_SERVICE_TYPE
+  );
 }
 
 export function isCrawlerIndex(
   index: ElasticsearchIndexWithIngestion | undefined
 ): index is CrawlerIndex {
-  return !!(index as CrawlerIndex)?.crawler;
+  const crawlerIndex = index as CrawlerIndex;
+  return (
+    !!crawlerIndex?.crawler ||
+    (!!crawlerIndex?.connector &&
+      crawlerIndex.connector.service_type === ENTERPRISE_SEARCH_CONNECTOR_CRAWLER_SERVICE_TYPE)
+  );
```

Comment:
We can remove these two lines. As this is what will break the types. If the `elastic-crawler` has the crawler key then that's enough to satisfy this function. and excluding `elastic-crawler` from returning as a `ConnectorIndex` should accomplish what you need.

```suggestion
  return (
    !!crawlerIndex?.crawler
  );
```

Reply:
@TattdCodeMonkey I understand the redundancy here so I've removed it, but out of curiosity how does this break type?

Reply:
Because of the `||` this will return `true` if given an object that has `crawler` or `connector` key, but the `CrawlerIndex` type must have a `crawler` key, and `connector` is optional. So if this was given an object without `crawler` but with `connector` that had `service_type === 'elastic_crawler'` it would pass.

Which probably wouldn't happen in practice, but this check would allow it to happen. Then that could cause a runtime exception since any code relying on this guard would expect `.crawler` to never be undefined, but this guard would technically allow it.